PSV3_1

Rocky640 opened the pull request:
Small speed-up in pawn.cpp
No functional change
Results for 10 tests for each version:

Base      Test      Diff
Mean    1435636   1445238   -9602
StDev   22576     23189     1848

p-value: 1
speedup: 0.007

DU-jdto commented:
I don't see how this could be non-functional with the backward pawn weights adjusted.

locutus2 commented:
In this form this is a functional change because you use other backward pawn values

Rocky640 commented:
In the current master code, a "backward" pawn is ALWAYS an  "unsupported" pawn (since we exclude connected pawns)

Here I changed "backward" penalties for ("backward" + "unsupported" S(20,10) )
and changed the logic to avoid the "unsupported" addition again.

So whatever bench you can think of, it will ALWAYS give the same result.

This is also consistent with what was done with the Isolated pawns (the unsupported weight was already included)
This also has the merit to illustrate more clearly that the overall penalty for a backward pawn is greater than for an Isolated pawn, which I find rather surprising especially in the opposed case. I'm currently doing some local tests on those weights.

mcostalba commented:
I confirm the small speed up on mingw gcc 4.9.2

Results for 10 tests for each version:
BaseTestDiffMean14578331460499-2666StDev74548751032606p-value: 84,70%
speedup: 0,20%

hxim commented:
My results:
Results for 20 tests for each version:

            Base      Test      Diff      
    Mean    1279453   1284523   -5070     
    StDev   1721      3313      2615      

p-value: 0,974

glinscott commented:
Cool, looks good.

Rocky640 commented:
A simplification test is about to be run. 
I suggest to wait for the result before committing this small speed-up.

mstembera commented:
Maybe too late but simplifications should always run against the best current known version.  So applying this first and then running the simplification against this would have been better.

zamar commented:
approved. This patch makes sense logically.
